src/gc.c

```diff
@@ -773,7 +772,20 @@ obj_free(mrb_state *mrb, struct RBasic *obj, mrb_bool end)
     {
       struct mrb_context *c = ((struct RFiber*)obj)->cxt;
 
-      if (c != mrb->root_c) {
+      if (c && c != mrb->root_c) {
```

Comment:
Since `mrb->root_c` is never NULL, I don't think the `c &&` check is necessary.

Reply:
Thanks for your review.

I checked and it seems that omitting the `NULL` check on the `c` variable causes a crash due to a "null pointer dereference".

```console
% lldb18 -- bin/mruby -e 'Fiber.allocate; GC.start'
(lldb) target create "bin/mruby"
Current executable set to '/var/tmp/mruby/bin/mruby' (x86_64).
(lldb) settings set -- target.run-args  "-e" "Fiber.allocate; GC.start"
(lldb) r
Process 90188 launched: '/var/tmp/mruby/bin/mruby' (x86_64)
Process 90188 stopped
* thread #1, name = 'mruby', stop reason = signal SIGSEGV: address not mapped to object (fault address: 0x30)
    frame #0: 0x0000000000413264 mruby`obj_free(mrb=0x000008d4df809000, obj=0x000008d4df82ad70, end=false) at gc.c:776:24
   773        struct mrb_context *c = ((struct RFiber*)obj)->cxt;
   774
   775        if (c != mrb->root_c) {
-> 776          if (!end && c->status != MRB_FIBER_TERMINATED) {
   777            mrb_callinfo *ci = c->ci;
   778            mrb_callinfo *ce = c->cibase;
   779
```

`mrb_free_context()` is "NULL safe", so there is no problem.

Reply:
Well, thank you for the confirmation. My bad.